modules/nexus/src/main/scala/ch/epfl/bluebrain/nexus/rdf/syntax/nexus.scala

```diff
@@ -0,0 +1,38 @@
+package ch.epfl.bluebrain.nexus.rdf.syntax
+
+import ch.epfl.bluebrain.nexus.rdf.Node.IriOrBNode
+import ch.epfl.bluebrain.nexus.rdf.syntax.node.unsafe._
+import ch.epfl.bluebrain.nexus.rdf.{Graph, Node}
+
+object nexus {
+
+  private final val rdfType = url"http://www.w3.org/1999/02/22-rdf-syntax-ns#type"
+
+  final implicit class NexusGraphOps(graph: Graph) {
+
+    /**
+      * @return The optionally available root ''subject'' of the Graph. This is, the subject which is not used as an object
+      */
+    def id: Option[IriOrBNode] =
```

Comment:
I don't think there's a need to subtract and collect twice; the resulting `IriOrBNode` handles both options and can be pattern matched for the two cases.